app/workers/deletes.php

```diff
@@ -307,24 +313,139 @@ protected function deleteAuditLogsByResource(string $resource, string $projectId
+                Console::success('Deleted deployment files: ' . $document->getAttribute('path', ''));
             } else {
-                Console::error('Failed to delete code deployment: ' . $document->getAttribute('path', ''));
+                Console::error('Failed to delete deployment files: ' . $document->getAttribute('path', ''));
             }
         });
 
+        /**
+         * Delete builds
+         */
+        if (!empty($deploymentIds)) {
+            $storageBuilds = new Local(APP_STORAGE_BUILDS . '/app-' . $projectId);
+            $this->deleteByGroup('builds', [
+                new Query('deploymentId', Query::TYPE_EQUAL, $deploymentIds)
+            ], $dbForProject, function (Document $document) use ($storageBuilds) {
+                if ($storageBuilds->delete($document->getAttribute('outputPath', ''), true)) {
+                    Console::success('Deleted build files: ' . $document->getAttribute('outputPath', ''));
+                } else {
+                    Console::error('Failed to delete build files: ' . $document->getAttribute('outputPath', ''));
+                }
+            });
+        }
+
         // Delete Executions
         $this->deleteByGroup('executions', [
             new Query('functionId', Query::TYPE_EQUAL, [$document->getId()])
         ], $dbForProject);
+
+    }
+
+    /**
+     * @param Document $document deployment document
+     * @param string $projectId
+     */
+    protected function deleteDeployment(Document $document, string $projectId): void
+    {
+        $dbForProject = $this->getProjectDB($projectId);
+
+        /**
+         * Request executor to delete the deployment containers
+         */
+        try {
+            $ch = \curl_init();
+            \curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'DELETE');
+            // TODO: Implement coroutines.
+            \curl_setopt($ch, CURLOPT_URL, "http://appwrite-executor/v1/deployments/{$document->getId()}");
+            \curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+            \curl_setopt($ch, CURLOPT_HTTPHEADER, [
+                'Content-Type: application/json',
+                'x-appwrite-project: '. $projectId,
+                'x-appwrite-executor-key: '. App::getEnv('_APP_EXECUTOR_SECRET', '')
+            ]);
+
+            $executorResponse = \curl_exec($ch);
+            $error = \curl_error($ch);
+            if (!empty($error)) {
+                throw new Exception($error, 500);
+            }
+
+            $statusCode = \curl_getinfo($ch, CURLINFO_HTTP_CODE);
+            if ($statusCode >= 400) {
+                throw new Exception('Executor error: ' . $executorResponse, $statusCode);
+            }
+
+            \curl_close($ch);
+        } catch (Throwable $th) {
+            Console::error($th->getMessage());
+        }
+
+        /**
+         * Delete deployment files
+         */
+        $storageFunctions = new Local(APP_STORAGE_FUNCTIONS . '/app-' . $projectId);
```

Comment:
Remember use of storage has to be refactored everywhere, once merged with the storage bucket changes.